src/parser.ts

```diff
@@ -943,26 +969,44 @@ export const ZodParser = (schema: z.ZodType<any>) => (
   } else {
     // if (params.async == true) {
     const checker = async () => {
-      const resolvedValue = await PROMISE.getValueAsync();
-
-      await Promise.all(
-        customChecks.map(async check => {
-          await check.check(resolvedValue, checkCtx);
-          // if (!checkResult) {
-          //   const { check: checkMethod, ...noMethodCheck } = check;
-          //   ERROR.addIssue(makeError(noMethodCheck));
-          // } else {
-          // }
-        }),
-      );
-
-      if (resolvedValue === INVALID && ERROR.isEmpty) {
-        ERROR.addIssue(
-          makeError({
-            code: ZodIssueCode.custom,
-            message: 'Invalid',
-          }),
-        );
+      let resolvedValue = await PROMISE.getValueAsync();
+ 
+      if (resolvedValue !== INVALID) {
+        let someError: boolean = false;
```

Comment:
I don't like that the custom checks short-circuit once an error is encountered. In general Zod tries to surface as many errors as possible upfront, instead of doling them out one by one. Is there some other reason you implemented it this way?

Reply:
My motivation here was async validations which hit a server, which are costly and I don't want to run them unless the sync validations succeed.  So (for example) if a username doesn't fit the expected regex, skip checking if the username already exists.

I see the value of surfacing all the errors up front.  Could we make it an optional setting, so it defaults to running them all in parallel, but we could opt-in to running them serially if we want?

Reply:
Sure I can implement this. How does `runAsyncValidationsInSeries` sound as a parameter name?